distributed/core.py

```diff
@@ -462,9 +462,6 @@ def send_recv_from_rpc(**kwargs):
             raise gen.Return(result)
         return send_recv_from_rpc
 
-    def __del__(self):
-        self.close_streams()
```

Comment:
Hmm, spawning asynchronous code from `__del__` is fraught with dangers (because it can revive the object being del'ed). I think we should therefore not resinstate those methods, and instead make sure we call the `close_rpc` or `close_streams` methods from owning objects.